docs/manual/macros.md

```diff
@@ -165,10 +165,12 @@ various common operations.
 
 ### Diagnostics
 
-| Macro          | Description
-| -------------- | -----------
-| `%trace`       | toggle print of debugging information before/after expansion
-| `%dump`        | print the active (i.e. non-covered) macro table
+| Macro            | Description | Introduced
+| ---------------- | ----------- | ----------
+| `%trace`         | toggle print of debugging information before/after expansion
+| `%dump`          | print the active (i.e. non-covered) macro table
+| `%__file_name`   | current file (if parsing a file) | 4.15
+| `%__file_lineno` | line in current file (if parsing a file) | 4.20
```

Comment:
%__file_lineno was introduced at the same time as __file___name, it's just that it's just much faster in 4.20.

Reply:
Also "line *number*" to make it clear it's not the line itself but its number.

Reply:
Looks like I read the patches wrong.